lua/plugins/persisted.lua

```diff
@@ -16,19 +16,32 @@ persisted.setup({
 	follow_cwd = false,
 	use_git_branch = false,
 	ignored_dirs = { '/tmp', '~/.cache' },
+})
+
+local group = vim.api.nvim_create_augroup("PersistedHooks", {})
 
-	before_save = function()
+vim.api.nvim_create_autocmd({ "User" }, {
+  pattern = "PersistedSavePre",
+  group = group,
+  callback = function()
 		require('interface').win.close_plugin_owned()
-	end,
+  end,
+})
+
+vim.api.nvim_create_autocmd({ "User" }, {
+  pattern = "PersistedTelescopeLoadPre",
+  group = group,
+  callback = function()
+		persisted.save()
+		persisted.stop()
+  end,
+})
 
-	telescope = {
-		before_source = function()
-	 		persisted.save()
-			persisted.stop()
-		end,
-		after_source = function(session)
-			persisted.start()
-			print('Started session ' .. session.name)
-		end
-	}
+vim.api.nvim_create_autocmd({ "User" }, {
+  pattern = "PersistedTelescopeLoadPre",
+  group = group,
+  callback = function(session)
+		persisted.start()
+		print('Started session ' .. session.name)
```

Comment:
We've changed this slightly in the new api and it should be `session.data.name` instead of `session.name`

Reply:
woah, a review from the author of the plugin themselves! It doesn't get better than this. Cheers mate!
```suggestion
		print('Started session ' .. session.data.name)
```